### PR TITLE
Update VK_PIPELINE_STAGE_ALL_GRAPHICS_BIT expansion

### LINKED ISSUE #695: VK_EXT_transform_feedback: Specifying VK_PIPELINE_STAGE_TRANSFORM_FEEDBACK_BIT_EXT in global mem barriers raises invalid validation errors
The error I'm seeing is:

```
Dependency 0 is a self-dependency, but specifies stage masks that contain stages not
in the GRAPHICS pipeline. The Vulkan spec states: If srcSubpass is equal to dstSubpass,
srcStageMask and dstStageMask must n ot set any bits that are neither 
VK_PIPELINE_STAGE_ALL_GRAPHICS_BIT, nor one of the graphics pipeline stages (https://www.khronos.org/registry/vulkan/specs/1.1-extensions/html/vkspec.html#VUID-
VkSubpassDependency2KHR-srcSubpass-02244)
```

This is seen when defining a self-subpass dependency necessary to separate separate draw call
invocations which are wrapped in begin+end transform feedback calls that use counter buffers.

Analogous error is reported for XFB-specific access masks.

### PR DESCRIPTION
**Pull request by jzulauf-lunarg**
With the spate of recent extensions the number of valid bits that `VK_PIPELINE_STAGE_ALL_GRAPHICS_BIT` expands into has changed.  Updated the expansion list from the Vulkan-Docs:chapters/synchronization.txt
list.  Expansion is conditional on enabled extensions.  Bits for non-enabled extensions will not be set during ALL_GRAPHICS_BIT expansion.

Additionally updated stageMask bit to stage ordinal list.

Fixes #695

### CODE REVIEW

**Comment by jzulauf-lunarg** [comment]
@mikes-lunarg @mark-lunarg  @tobine  -- want to make sure you concur on gating the bit expansion on the extension enabling.  Each of the new bits in the spec is ifdef'd with the expansion name, so it would seem to me that these bits aren't valid to infer without the matching extension being enabled.

Also, I wouldn't want any post expansion checks to complain about bits being set if checked against the expanded set

File: layers/core_validation.cpp [hunk 1]
```diff
@@ -7543,17 +7543,21 @@ void PreCallRecordCmdResetEvent(VkCommandBuffer commandBuffer, VkEvent event, Vk
...
 
     return (inflags & ~VK_PIPELINE_STAGE_ALL_GRAPHICS_BIT) |
-           (VK_PIPELINE_STAGE_TOP_OF_PIPE_BIT | VK_PIPELINE_STAGE_DRAW_INDIRECT_BIT | VK_PIPELINE_STAGE_VERTEX_INPUT_BIT |
-            VK_PIPELINE_STAGE_VERTEX_SHADER_BIT | VK_PIPELINE_STAGE_TESSELLATION_CONTROL_SHADER_BIT |
-            VK_PIPELINE_STAGE_TESSELLATION_EVALUATION_SHADER_BIT | VK_PIPELINE_STAGE_GEOMETRY_SHADER_BIT |
-            VK_PIPELINE_STAGE_FRAGMENT_SHADER_BIT | VK_PIPELINE_STAGE_EARLY_FRAGMENT_TESTS_BIT |
-            VK_PIPELINE_STAGE_LATE_FRAGMENT_TESTS_BIT | VK_PIPELINE_STAGE_COLOR_ATTACHMENT_OUTPUT_BIT |
-            VK_PIPELINE_STAGE_BOTTOM_OF_PIPE_BIT | VK_PIPELINE_STAGE_SHADING_RATE_IMAGE_BIT_NV |
-            VK_PIPELINE_STAGE_TASK_SHADER_BIT_NV | VK_PIPELINE_STAGE_MESH_SHADER_BIT_NV);
+           (VK_PIPELINE_STAGE_TOP_OF_PIPE_BIT | VK_PIPELINE_STAGE_DRAW_INDIRECT_BIT |
```
Comment (mikes-lunarg) [inline layers/core_validation.cpp]: :+1: for maintaining the enum order from the synchronization chapter

**Comment by DominikWitczakAMD** [comment]
These changes do reduce the number of errors I'm seeing when using XFB. However, I'm still seeing the following error being reported when doing a TRANSFORM_FEEDBACK_BIT_EXT::TRANSFORM_FEEDBACK_COUNTER_WRITE_BIT_EXT->TRANSFORM_FEEDBACK_COUNTER_READ_BIT_EXT global mem barrier.

**Comment by jzulauf-lunarg** [comment]
@DominikWitczakAMD  -- looks like your last message got cut off.  What is the error message you're seeing?

File: layers/core_validation.cpp [hunk 2]
```diff
@@ -7562,18 +7566,35 @@ static bool HasNonFramebufferStagePipelineStageFlags(VkPipelineStageFlags inflag
 }
 
 static int GetGraphicsPipelineStageLogicalOrdinal(VkPipelineStageFlagBits flag) {
+    // Note that the list (and lookup) ignore invalid-for-enabled-extension condition.  This should be checked elsewhere
```
Comment (jzulauf-lunarg) [inline layers/core_validation.cpp]: Yet another list that reflects the valid stages... @mikes-lunarg @DominikWitczakAMD  -- likely matches the additional errors seen with the former push.

File: layers/core_validation.cpp [hunk 3]
```diff
@@ -7562,18 +7566,35 @@ static bool HasNonFramebufferStagePipelineStageFlags(VkPipelineStageFlags inflag
...
-        VK_PIPELINE_STAGE_TESSELLATION_EVALUATION_SHADER_BIT, VK_PIPELINE_STAGE_GEOMETRY_SHADER_BIT,
+        VK_PIPELINE_STAGE_TOP_OF_PIPE_BIT,
+        VK_PIPELINE_STAGE_DRAW_INDIRECT_BIT,
+        VK_PIPELINE_STAGE_VERTEX_INPUT_BIT,
+        VK_PIPELINE_STAGE_VERTEX_SHADER_BIT,
+        VK_PIPELINE_STAGE_TESSELLATION_CONTROL_SHADER_BIT,
+        VK_PIPELINE_STAGE_TESSELLATION_EVALUATION_SHADER_BIT,
+        VK_PIPELINE_STAGE_GEOMETRY_SHADER_BIT,
+        VK_PIPELINE_STAGE_TRANSFORM_FEEDBACK_BIT_EXT,
+        VK_PIPELINE_STAGE_SHADING_RATE_IMAGE_BIT_NV,
```
Comment (mikes-lunarg) [inline layers/core_validation.cpp]: VK_PIPELINE_STAGE_SHADING_RATE_IMAGE_BIT_NV is after VK_PIPELINE_STAGE_MESH_SHADER_BIT_NV in the "graphics mesh shading pipeline" list

Comment (jzulauf-lunarg) [reply layers/core_validation.cpp]: Fixed. Pushed.

**Pull request merged**